### PR TITLE
(#1107) Create nci header default source

### LINKED ISSUE #1106: Enabler: Set up default adapter for Mega Menu sources
## Create default mobile menu source

We need to create a Mega Menu source that does not break the main menu. The DS cannot assume content for the mega menu so a custom source is required for the NCI Extended Header w/ Mega Menu component. However, we can use a default source for the auto init for 

Dev notes:
- Return `null` or `''`, whatever does not break the mega menu code
- `console.warn` user that a custom source was not specified

### Related
- https://github.com/NCIOCPL/ncids/issues/1108
- #1157
- https://github.com/NCIOCPL/ncids/issues/1107

### LINKED ISSUE #1107: Enabler: Create a Mobile Menu source that uses the primary navigation to drive the items in mobile menu
## Create default mobile menu source 

The default mobile menu source should take items from the primary navigation in the header and create basic links to populate the mobile menu. This will be used for auto initialization of the component. 

For MVP this should only take the primary nav items and create first-level links only without any children.

### Related

- #1108 
- #1157 
- #1106

### LINKED ISSUE #1157: Create auto init script for the header component
## Create auto init script for the header component 

Create auto init script for header component. Should use default sources for the mega and mobile menus as outlined in related tickets. Does not include autocomplete at this time.

### Tasks

- [x] Create auto init module
- [x] Update example js site

### Related

- #1108 
- #1107 
- #1106

### PR DESCRIPTION
**Pull request by olitharp-nci**
(#1107) Create nci header mobile menu default source
    
Closes #1107

---

(#1106) Create mega menu default source
    
- Output warning for missing custom mega menu source
- Update mega menu adaptor to allow for null sources for auto init
    
Closes #1106

---

(#1157) Create auto init script for nci header

- Updated babelrc, removed babel options from webpack
- Re-organize header variant export

Closes #1157

### CODE REVIEW

File: packages/ncids-js/src/components/nci-header/utils/mega-menu/mega-menu-nav.ts [hunk 1]
```diff
@@ -440,24 +440,24 @@ export class MegaMenuNav {
...
-		this.content.ariaAtomic = 'true';
-
-		// Replace loading container with menu
-		this.loader.classList.add('hidden');
-		this.loaderContainer.classList.add('hidden');
-		this.loaderContainer.replaceWith(this.content);
-
-		// Set button aria-controls to new content
-		button.setAttribute('aria-controls', id);
+			// Programmatically create unique id
```
Comment (bryanpizzillo) [inline packages/ncids-js/src/components/nci-header/utils/mega-menu/mega-menu-nav.ts:444]: There seems to be a need for a update of the unit tests for mega-menu? I assume this is being done in the if to not draw the item if it is null. Although, I don't know where it is drawing either.

File: packages/ncids-js/src/index.ts [hunk 2]
```diff
@@ -7,6 +7,9 @@ export { NCISubscribe } from './components/nci-subscribe/nci-subscribe.component
 import { NCIExtendedHeaderWithMegaMenu } from './components/nci-header/index';
 export { NCIExtendedHeaderWithMegaMenu };
 
+import { NCIExtendedHeaderWithMegaMenuAutoInit } from './components/nci-header/index';
```
Comment (bryanpizzillo) [inline packages/ncids-js/src/index.ts:10]: This should not be in the index. If you:
```
import { NCIExtendedHeaderWithMegaMenu } from './components/nci-header';
```
In something like CGDP, we will be bundling the index.js module. Which means we should also be getting the `NCIExtendedHeaderWithMegaMenuAutoInit` pulled along.

I was thinking that others would do something like:
```
import { NCIExtendedHeaderWithMegaMenu } from '@nciocpl/ncids-js/nci-header/auto-init';
```

File: testing/ncids-js-testing/src/auto-init.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import './index.scss';
+
+import { NCIExtendedHeaderWithMegaMenuAutoInit } from '@nciocpl/ncids-js';
+new NCIExtendedHeaderWithMegaMenuAutoInit();
```
Comment (bryanpizzillo) [inline testing/ncids-js-testing/src/auto-init.ts:4]: The idea with the autoinit is that you do not have to execute it. The import should be all you need.

File: packages/ncids-js/src/components/nci-header/utils/auto-init/index.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+				megaMenuSource: new DefaultMegaMenuSource(),
+				mobileMenuSource: new DefaultMobileMenuSource(),
+			});
+		}
+	}
+}
+
+/**
+ * Auto initialize component.
+ * new NCIExtendedHeaderWithMegaMenuAutoInit();
```
Comment (bryanpizzillo) [inline packages/ncids-js/src/components/nci-header/utils/auto-init/index.ts:33]: This should be called in this module.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Ok, let's talk tomorrow. There are a few things about the autoinit that is slightly off. Also I have a genuine question about the mega menu source.

**Review by karlikpj — APPROVED** [review]

File: testing/ncids-js-testing/public/example-pages/auto-init-example.html [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+					</form>
+				</div>
+			</div>
+
+			<nav aria-label='Primary navigation' class='nci-header-nav'>
+				<div class='nci-header-nav__inner'>
+					<ul class='nci-header-nav__primary'>
+						<li class='nci-header-nav__primary-item'>
+							<a
+								href='./../assets/data/mega-menu/example-data-1.html'
```
Comment (bryanpizzillo) [inline testing/ncids-js-testing/public/example-pages/auto-init-example.html:142]: Can we either use `javascript:void();` or `#` here?

File: testing/ncids-js-testing/src/auto-init.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+import '@nciocpl/ncids-js/components/nci-header/auto-init/index';
```
Comment (bryanpizzillo) [inline testing/ncids-js-testing/src/auto-init.ts:1]: Can't you just drop the `/index` here?

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Also, can you creatively organize this into 
* `src/components/nci-header/extended-with-mega-menu/index.ts` - the current header export.
* `src/components/nci-header/extended-with-mega-menu/auto-init/index.ts` - the auto init.

So you would do:
`import header from '@nciocpl/ncids-js/components/nci-header/extended-with-mega-menu`
and
`import '@nciocpl/ncids-js/components/nci-header/auto-init/extended-with-mega-menu';`

**Review by dlescarbeau — APPROVED** [review]

**Review by bryanpizzillo — APPROVED** [review]
Looks good.
